pkg/authorization/toplevel_org_authorizer.go

```diff
@@ -105,15 +108,17 @@ func (a *topLevelOrgAccessAuthorizer) Authorize(ctx context.Context, attr author
 	}
 
 	// check the org workspace exists in the root workspace
-	if _, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(tenancyv1alpha1.RootCluster, requestTopLevelOrgName)); err != nil {
+	topLevelWSKey := clusters.ToClusterAwareKey(tenancyv1alpha1.RootCluster, requestTopLevelOrgName)
+	if _, err := a.clusterWorkspaceLister.Get(topLevelWSKey); err != nil {
 		if errors.IsNotFound(err) {
+			kaudit.AddAuditAnnotation(ctx, AuditAnnotationWorkspaceAccessNotPermittedDetails, fmt.Sprintf("clusterworkspace %q not found", topLevelWSKey))
 			return authorizer.DecisionDeny, WorkspaceAcccessNotPermittedReason, nil
 		}
+		kaudit.AddAuditAnnotation(ctx, AuditAnnotationWorkspaceAccessNotPermittedDetails, fmt.Sprintf("error getting clusterworkspace %q: %v", topLevelWSKey, err))
 		return authorizer.DecisionNoOpinion, WorkspaceAcccessNotPermittedReason, err
 	}
 
-	switch {
-	case isServiceAccount:
+	if isServiceAccount {
```

Comment:
don't like this rewrite. We intentionally wrote it with a switch to be it clearer. It's not obvious that `isUser != isServiceAccount` but rather a coincidence of the current setup.

Reply:
yes agreed, please leave the switch.

Reply:
Ok..... but it does make returning harder